crates/fuel-core/src/service/genesis.rs

```diff
@@ -151,150 +147,146 @@ fn import_genesis_block(
 
 fn init_coin_state(
     db: &mut Database,
-    state: &Option<StateConfig>,
+    state: &StateConfig,
+    height: Option<BlockHeight>,
```

Comment:
Just curious, why did the height get pulled out of the `StateConfig`?

Reply:
It doesn't make a big difference in the current PR but the final goal is for the state config to be consumed as a stream because of memory limitations. 
During the streaming process, we make use of the height to validate tx pointers. Having it as a par of the chain config is more convenient since it's loaded into memory before we start streaming.